### PR TITLE
Better support for IStructuralEquatable under NUnit

### PR DESCRIPTION
**Pull request by marklam**
Some objects that implement `IStructuralEquatable` (eg. [`System.Collections.Immutable.ImmutableArray`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutablearray-1?view=netcore-3.1)) don't currently compare equal in FsUnit.Nunit in the same way as using F#'s operator equals.

XUnit and MSTest don't seem to be affected.

This change adds equality constraints that will fall back to structural equality & comparison when required. These constraints are used by FsUnit, FsUnitTyped and GenericAssert.

There are tests to check that all three unit test platforms behave the same, and new tests to cover GenericAssert.

Also Assert.Contains is updated to remove a deprecated API warning from NUnit.

### CODE REVIEW

File: src/FsUnit.NUnit/FsUnit.fs [hunk 1]
```diff
@@ -45,7 +45,7 @@ module TopLevelOperators =
         else
             Assert.That(y, c)
 
-    let equal x = EqualConstraint(x)
+    let equal x = Equality.IsEqualTo(x)
```
Comment (sergey-tihon) [inline src/FsUnit.NUnit/FsUnit.fs:48]: This is breaking changes, if we merge this we will not be able use `FsUnit.fs` using paket (without referencing NuGet package)
- http://fsprojects.github.io/FsUnit/Paket.html
- http://fsprojects.github.io/FsUnit/FsUnitTyped.html#FsUnitTyped-with-Paket

Comment (marklam) [reply src/FsUnit.NUnit/FsUnit.fs:48]: So this type would need to be moved/copied into FsUnit.fs and FsUnitTyped.fs ?

Comment (sergey-tihon) [reply src/FsUnit.NUnit/FsUnit.fs:44]: Just double-check `Lightweight FsUnit with Paket`  guide and it is already broken by this line -https://github.com/fsprojects/FsUnit/blob/master/src/FsUnit.NUnit/FsUnit.fs#L6

**Comment by sergey-tihon** [comment]
@marklam Thank you for this PR!

**Pull request merged**